Framework/API/src/MultiPeriodGroupWorker.cpp

```diff
@@ -203,12 +203,20 @@ bool MultiPeriodGroupWorker::processGroups(
   WorkspaceGroup_sptr outputWS = boost::make_shared<WorkspaceGroup>();
   AnalysisDataService::Instance().addOrReplace(outName, outputWS);
 
+  double progress_proportion = 1.0 / static_cast<double>(nPeriods);
   // Loop through all the periods. Create spawned algorithms of the same type as
   // this to process pairs from the input groups.
   for (size_t i = 0; i < nPeriods; ++i) {
     const int periodNumber = static_cast<int>(i + 1);
-    Algorithm_sptr alg_sptr = API::AlgorithmManager::Instance().createUnmanaged(
-        sourceAlg->name(), sourceAlg->version());
+    // use create Child Algorithm that look like this one
+    Algorithm_sptr alg_sptr = sourceAlg->createChildAlgorithm(
+        sourceAlg->name(), progress_proportion * periodNumber,
+        progress_proportion * (1 + periodNumber), sourceAlg->isLogging(),
+        sourceAlg->version());
+    // Don't make the new algorithm a child so that it's workspaces are stored
+    // correctly
+    alg_sptr->setChild(false);
+    alg_sptr->setRethrows(true);
```

Comment:
I see the point in creating the algorithm with `createChildAlgorithm` so the behavior of the progress bar is adequate. But with this we have one more place with the funny sequence of events whereby a child algorithm is ripped off its childhood!
Would `setAlwaysStoreInADS(false)` serve the same purpose as `setChild(false)` and be a bit more clear? Or would there be any undesired difference? As far as I can see what `m_isChild` brings in addition to `m_alwaysStoreInADS` is that it:
- suppresses some logs
- disables workspace locking